packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationFirebaseMessagingService.kt

```diff
@@ -84,4 +92,16 @@ class PushNotificationFirebaseMessagingService : FirebaseMessagingService() {
             }
         }
     }
+
+    private fun runAppFromKilledState(){
+        // Check if there is already a main Flutter Engine running
+        val mainEngine = FlutterEngineCache.getInstance()
+            .get(PushNotificationPluginConstants.FLUTTER_ENGINE_ID)
+        if (mainEngine == null) {
+            StreamHandlers.deInit()
```

Comment:
Is it possible that `StreamHandlers` could have been `init`'ed when the App is at the killed state?

Reply:
Since I added the refresh flag to `StreamHandlers` init method and use that in `onAttachedToEngine`, this is not required. Removing it. 
The `StreamHandlers` are `init'ed` when the app receives a notification in the killed state and user opens the app. Hence we refresh the handlers so they are attached to the main engine.